lib/cancan/rule.rb

```diff
@@ -142,7 +142,11 @@ def model_adapter(subject)
       CanCan::ModelAdapters::AbstractAdapter.adapter_class(subject_class?(subject) ? subject : subject.class)
     end
 
-    def condition_match?(attribute, value)
+    def condition_match?(subject, name, value)
+      attribute = subject.send(name)
+
+      return value.where(name => attribute).any? if defined?(ActiveRecord) && value.is_a?(ActiveRecord::Relation)
```

Comment:
@laertispappas The users authorised to read should not be passed in the collection. A collection of records which are allowed to be read should be returned.

So lets say that a user should only be able to read the articles that he wrote himself.

```ruby
can :read, Article, id: Article.where(user_id: @user.id)
```

Reply:
I see your point. Can't a custom rule with block help you achieve that?

```ruy
can :read, Article do |article|
  article.user_id == @user.id
end
```

Reply:
@laertispappas I don't believe blocks work with `accessible_by` and it couldn't make use of existing database scopes either. If you used relationships, I guess it would also result in a lot of separate database queries.

Some of the pros of using a collection would be, that it did everything in one database query and that it would work with both `can?`, `authorize`, `accessible_by` and `load_and_authorize` etc.